### ISSUE TITLE
Additional option support for Database instance creation

### ISSUE THREAD

**Issue by tomsanbear** [motherduckdb/duckdb-async#21]
Thanks for this library, saved me some time from having to promisify the original callback based library.

One thing missing however is full support of the options supported in the original database constructor (https://github.com/duckdb/duckdb-node/blob/55a3602f42a0019b25d2ec9c48b3c6e6bf55802f/lib/duckdb.d.ts#L147C7-L147C7). I'd love to contribute this code to the repository if you are open, but didn't see any contribution guidelines. If there aren't any specific guidellines then I can probably open an PR for this later this week.

Thanks!

**Comment by jlarmstrongiv** [comment]
That’d be fantastic!  I am running into out of memory errors, because I cannot configure database options

```js
const db = new duckdb.Database(':memory:', {
    "access_mode": "READ_WRITE",
    "max_memory": "512MB",
    "threads": "4"
}, (err) => {
  if (err) {
    console.error(err);
  }
});
```

**Comment by jlarmstrongiv** [comment]
Some options can’t be changed with `SET` statements while running either

```
ERROR	[Error: Invalid Input Error: Cannot change allow_unsigned_extensions setting while database is running] {
  errno: -1,
  code: 'DUCKDB_NODEJS_ERROR',
  errorType: 'Invalid Input'
}
```

**Comment by jlarmstrongiv** [comment]
It looks like the options are passed straight through to `duckdb`, so using a `// @ts-expect-error` will work

```
  // @ts-expect-error Argument of type 'Config' is not assignable to parameter of type 'number'.
  const db = await Database.create(filePath, config);
```

**Comment by antonycourtney** [comment]
Hi folks, thanks for surfacing this!  Looks like the type of the `Database` constructor was widened in the underlying node bindings in a recent DuckDb release. I've started a `duckdb-latest` branch in this repository and just committed a fix to address this issue.  Will promote this branch (and publish it to npm) when the node bindings for 0.10.0 are published.
Thanks for reporting the issue, the constructive interim workarounds, and for your patience!

**Comment by tgdn** [comment]
Any updates on this?

**Comment by elefeint** [comment]
This was fixed back in 0.10.2, so passing the full configuration dictionary works now, consistent with `duckdb-node`:
```
const db = await Database.create(':memory:', {
    "access_mode": "READ_WRITE",
    "max_memory": "512MB",
    "threads": "12"
}, (err) => {
    if (err) {
        console.error(err);
    }
});
```
